config/peerpods/podvm/podvm-builder.sh

```diff
@@ -87,9 +86,28 @@ function create_podvm_image() {
 # Function to delete podvm image
 # IMAGE_ID or AMI_ID is the input and expected to be set
 # These are checked in individual cloud provider scripts and if not set, the script will exit
+# Accepts two optional arguments
+# -f : force delete the image
+# -g : delete the image gallery
 
 function delete_podvm_image() {
 
+  local max_args=3
+  if (("${#@}" > max_args)); then
+    error_exit "Too many arguments passed to delete_podvm_image (max ${max_args} allowed)"
```

Comment:
Why checking the number of arguments ?

Reply:
This is to ensure the function is used in the intended way with the required number of arguments. Ideally I wanted to have a similar check for all the functions to ensure only the accepted number of arguments are passed but that's best suited for another PR.
If this doesn't make sense I'm happy to remove it.

Reply:
@gkurz can we move ahead with this or you want this check to be removed ?

Reply:
FWIW usual scheme is to check that the function is passed arguments it understands rather than the number of arguments.